packages/adapter-drizzle/src/planetscale/index.ts

```diff
@@ -0,0 +1,248 @@
+  schema?: Partial<CustomSchema>
+): Adapter {
+  const { users, accounts, sessions, verificationTokens } = {
+    users: schema?.users ?? defaultSchema.users,
+    accounts: schema?.accounts ?? defaultSchema.accounts,
+    sessions: schema?.sessions ?? defaultSchema.sessions,
+    verificationTokens:
+      schema?.verificationTokens ?? defaultSchema.verificationTokens,
+  }
+
+  return {
+    createUser: async (data) => {
+      const id = crypto.randomUUID()
+
+      await client.insert(users).values({ ...data, id })
+
+      return client
+        .select()
+        .from(users)
+        .where(eq(users.id, id))
+        .then((res) => res[0])
+    },
+    getUser: async (data) => {
+      return (
+        client
+          .select()
+          .from(users)
+          .where(eq(users.id, data))
+          .then((res) => res[0]) ?? null
+      )
+    },
+    getUserByEmail: async (data) => {
+      return (
+        client
+          .select()
+          .from(users)
+          .where(eq(users.email, data))
+          .then((res) => res[0]) ?? null
+      )
+    },
+    createSession: async (data) => {
+      await client.insert(sessions).values(data)
+
+      return client
+        .select()
+        .from(sessions)
+        .where(eq(sessions.sessionToken, data.sessionToken))
+        .then((res) => res[0])
+    },
+    getSessionAndUser: async (data) => {
+      return (
+        client
+          .select({
+            session: sessions,
+            user: users,
+          })
+          .from(sessions)
+          .where(eq(sessions.sessionToken, data))
+          .innerJoin(users, eq(users.id, sessions.userId))
+          .then((res) => res[0]) ?? null
+      )
+    },
+    updateUser: async (data) => {
+      if (!data.id) {
+        throw new Error("No user id.")
+      }
+
+      await client.update(users).set(data).where(eq(users.id, data.id))
+
+      return client
+        .select()
+        .from(users)
+        .where(eq(users.id, data.id))
+        .then((res) => res[0])
+    },
+    updateSession: async (data) => {
+      await client
+        .update(sessions)
+        .set(data)
+        .where(eq(sessions.sessionToken, data.sessionToken))
+
+      return client
+        .select()
+        .from(sessions)
+        .where(eq(sessions.sessionToken, data.sessionToken))
+        .then((res) => res[0])
+    },
+    linkAccount: async (rawAccount) => {
+      await client
+        .insert(accounts)
+        .values(rawAccount)
+        .then((res) => res.rows[0])
+    },
+    getUserByAccount: async (account) => {
+      const dbAccount =
+        await client
+          .select()
+          .from(accounts)
+          .where(
+            and(
+              eq(accounts.providerAccountId, account.providerAccountId),
+              eq(accounts.provider, account.provider)
+            )
+          )
+          .leftJoin(users, eq(accounts.userId, users.id))
+          .then(res => res[0])
+
+      return dbAccount.users
```

Comment:
Returning the query result is causing an Callback error, in one edge case:
- If user already exists but has no linked oauth account and tries to sign in

This check before returning solves it: 
`
if (!dbAccount) return null
`